nilearn/experimental/surface/maskers.py

```diff
@@ -297,9 +298,7 @@ def inverse_transform(self, masked_img: np.ndarray) -> SurfaceImage:
 
     def generate_report(self):
         """Generate a report."""
-        try:
-            from nilearn.reporting.html_report import generate_report
-        except ImportError:
+        if not is_matplotlib_installed():
             with warnings.catch_warnings():
                 mpl_unavail_msg = (
                     "Matplotlib is not imported! "
```

Comment:
It may be possible to refactor those indeed.
I have not looked into it. But that would be for another issue / set of PRs.

Given that I started some work to "uniformize" the reporting across maskers and GLM, I would be in favor of this.